extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -465,16 +504,42 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.updateInViewport(dev().assertElement(this.container_), true);
     this.scheduleLayout(dev().assertElement(this.container_));
 
-    this.currentElementId_ = element.lightboxItemId;
-    // Hack to access private property. Better than not getting
-    // type checking to work.
-    /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-        this.currentElementId_);
-
     this.win.document.documentElement.addEventListener(
         'keydown', this.boundHandleKeyboardEvents_);
 
-    return Promise.resolve();
+    if (this.carousel_) {
+      return this.resources_.requireLayout(this.carousel_)
+          .then(() => {
+            this.currentElemId_ = element.lightboxItemId;
+            // Hack to access private property. Better than not getting
+            // type checking to work.
+            /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
+                this.currentElemId_);
+            this.resizeImageViewerDimensions_();
+            this.updateDescriptionBox_();
+          });
+    } else {
```

Comment:
This if / else block was added to make type check happy. I'm not entirely sure what to do under this else block, since we should never actually end up traversing this branch of code. I followed some patterns that I'm seeing elsewhere and logging an error to `dev()`, and returning a `Promise.resolve()` to satisfy type check.

Reply:
removing the `if` and doing `dev().assert(this.carousel_)` before accessing it is the right way to handle "should never happen or is a bug" cases.